### ISSUE TITLE
Spin attack hitbox is uncomfortably small

### ISSUE THREAD

**Issue by Koopa1018** [GTcreyon/SM63Redux#118]
**Describe the bug**
Spinning hits enemies from an uncomfortably small range away.

The player's normal hitbox is slightly narrower than the player sprite when standing.  During a spin, the sprite's silhouette becomes several pixels wider, but the hitbox remains narrow, leading to cognitive dissonance when Mario's arms visibly touch the enemy, but nothing happens.

This is in contrast with the Flash version, wherein the spin attack hitbox was almost excessively generous.

**To Reproduce**
Our established muscle memory may obscure the problem. Recommend trying these steps on a fresh individual.
1. Allow a goomba to charge you.
2. Time a spin attack using only visual information (the default for many players)

**Expected behavior**
Spin attack range matches or exceeds what the player can see.

**Screenshots**
Arm visibly hitting the box, yet the box is unbroken.
![image](https://user-images.githubusercontent.com/29995463/227748490-b78bf710-c971-4a47-b03c-57a7cb4120ce.png)

**System Information**
 - OS/Browser: Win10
 - Game version: 0.1.6 alpha
 - Other specs: N/A

**Additional context**
Would recommend extending the hitbox past the visible boundaries slightly. as an affordance to players. (Though maybe not quite as much as the Flash version--too much of a good thing....)

Perhaps don't fix until the new spin sprites drop.

**Comment by Kuma-Boo** [comment]
Maybe instead of having two separate hitbox sizes, we can just have a timer that keeps track of how long the player's been spinning for.

That way, environmental objects like this will always consistently break, while enemies can check the timer to see if they should take damage or not.

**Comment by Koopa1018** [comment]
Nah, it's not a box-specific problem. (Boxes were just easy to screenshot acting weird, since they don't move.) Affecting enemies is what I'm going for here.

**Comment by Koopa1018** [comment]
Notes on Flash version's spin, copied from Kuma's comment in #223:

> After making some comparisons with the original flash version, The Flash version has an absurdly large hitbox, but seems to only be active for about 12 frames. Redux's spin in comparison has a super long active window of 30 frames.
> 
> Not really a big deal as we can always nerf it later and it's pretty unrelated to this pr, but I just thought I should document it.

**Comment by GTcreyon** [comment]
> The Flash version has an absurdly large hitbox, but seems to only be active for about 12 frames. Redux's spin in comparison has a super long active window of 30 frames.

Remember that the original 63 runs at 32fps, while Redux runs at 60fps. So, 12 frames in the original accounts for about 22.5 frames in Redux.

Still, 30 is longer than it's meant to be. I'll be doing the spin attack in the player rewrite now, so I'll bear this in mind.
